File: src/entities/docker.ts
export interface DockerSystemSummary {
  images_size: string;
  images_reclaimable: string;
  containers_size: string;
  containers_reclaimable: string;
  volumes_size: string;
  volumes_reclaimable: string;
}

export interface DockerContainerInfo {
  id: string;
  name: string;
  image: string;
  status: string;
  is_act: boolean;
}

export interface DockerImageInfo {
  id: string;
  repository: string;
  tag: string;
  size: string;
  is_act: boolean;
}

export interface DockerVolumeInfo {
  name: string;
  is_act: boolean;
}

File: src-tauri/src/domain/docker.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct DockerSystemSummary {
    pub images_size: String,
    pub images_reclaimable: String,
    pub containers_size: String,
    pub containers_reclaimable: String,
    pub volumes_size: String,
    pub volumes_reclaimable: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DockerContainerInfo {
    pub id: String,
    pub name: String,
    pub image: String,
    pub status: String,
    pub is_act: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DockerImageInfo {
    pub id: String,
    pub repository: String,
    pub tag: String,
    pub size: String,
    pub is_act: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DockerVolumeInfo {
    pub name: String,
    pub is_act: bool,
}
